colossalai/context/moe_context.py

```diff
@@ -0,0 +1,139 @@
+import torch
+import torch.distributed as dist
+from .parallel_mode import ParallelMode
+
+
+class MoeInfo:
+    """Moe parallelism information, storing parallel sizes and groups.
+    """
+
+    def __init__(self, ep_size: int, dp_size: int):
+        self.ep_size = ep_size
+        self.dp_size = dp_size
+        self.ep_group = None
+        self.dp_group = None
+
+        # Here we assume tensor parallel size = 1
+        # Otherwise, MoE can't be used
+        # Since TENSOR parallel group and DATA parallel group
+        # have been created, we can use them directly.
+        if ep_size == 1:
+            from colossalai.core import global_context as gpc
+            self.ep_group = gpc.get_group(ParallelMode.TENSOR)
+            self.dp_group = gpc.get_group(ParallelMode.DATA)
+            return
+
+        if dp_size == 1:
+            from colossalai.core import global_context as gpc
+            self.ep_group = gpc.get_group(ParallelMode.DATA)
+            self.dp_group = gpc.get_group(ParallelMode.TENSOR)

```

Comment:
so wired....
self.dp_group is diff from gpc's DP group
Add comments to explain what is self.dp_group